rclcpp/src/rclcpp/client.cpp

```diff
@@ -78,12 +84,6 @@ ClientBase::wait_for_service_nanoseconds(std::chrono::nanoseconds timeout)
     // check was non-blocking, return immediately
     return false;
   }
-  // make an event to reuse, rather than create a new one each time
-  auto node_ptr = node_graph_.lock();
-  if (!node_ptr) {
-    throw InvalidNodeError();
-  }
-  auto event = node_ptr->get_graph_event();
```

Comment:
For time travelers, this block was moved above the `if (this->service_is_ready()) {return true;}` block on line 79. The reason being is that if the graph changed between that call on line 79 and here it could cause the state change of `this->service_is_ready()` to be missed.